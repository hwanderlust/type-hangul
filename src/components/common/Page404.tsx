import React from "react";
import { Link, useLocation } from "react-router-dom";
import styled from "styled-components";

import { Fonts, Sizes } from "../../helpers";

const Main = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
`;
const List = styled.ul`
  margin: 0;
  padding: 0;
`;
const Item = styled.li`
  font-size: ${Sizes.variable.font.medium};
`;
const StyledLink = styled(Link)`
  text-decoration: none;
  color: black;
  font-family: ${Fonts.playfair};

  &:focus {
    text-decoration: underline;
    text-decoration-color: black;
    outline: none;
  }

  &:hover {
    color: white;
  }
`;

function Page404() {
  return (
    <Main>
      <div>
        <h1>Oops! Looks like you're a bit lost.</h1>
        <p>"<strong>{useLocation().pathname}</strong>" isn't a path we have on this site.</p>
        <h2>If you're here, you're here for the games! Please have a gander.</h2>
        <List>
          <Item><StyledLink to="/game/pop">Pop</StyledLink></Item>
          <Item><StyledLink to="/game/jump">Jump</StyledLink></Item>
        </List>
      </div>
    </Main>
  );
}

export default Page404;