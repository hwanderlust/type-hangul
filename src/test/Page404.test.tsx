import 'jest-styled-components';
import React from 'react';
import { BrowserRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

import Page404 from "../components/common/Page404";

describe("<Page404 />", () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Page404 />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
