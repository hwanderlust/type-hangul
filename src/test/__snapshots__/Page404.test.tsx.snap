// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page404 /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100vh;
}

.c1 {
  margin: 0;
  padding: 0;
}

.c2 {
  font-size: calc(24px + (36 - 24) * ((100vw - 300px) / (1440 - 300)));
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
  font-family: 'Playfair Display',serif;
}

.c3:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-color: black;
  text-decoration-color: black;
  outline: none;
}

.c3:hover {
  color: white;
}

<main
  className="c0"
>
  <div>
    <h1>
      Oops! Looks like you're a bit lost.
    </h1>
    <p>
      "
      <strong>
        /
      </strong>
      " isn't a path we have on this site.
    </p>
    <h2>
      If you're here, you're here for the games! Please have a gander.
    </h2>
    <ul
      className="c1"
    >
      <li
        className="c2"
      >
        <a
          className="c3"
          href="/game/pop"
          onClick={[Function]}
        >
          Pop
        </a>
      </li>
      <li
        className="c2"
      >
        <a
          className="c3"
          href="/game/jump"
          onClick={[Function]}
        >
          Jump
        </a>
      </li>
    </ul>
  </div>
</main>
`;
